WordPress/gutenberg
# Site editor sidebar: refactor page details

## What?

The redesign for the site editor sidebar's individual page details were introduced in https://github.com/WordPress/gutenberg/pull/50767.

This PR refactors the code by:

1. rolling back exposing `getMediaDetails` from the editor package and using fetched attachment media props instead
2. refactoring page details and the useSelect callbacks
3. moving the generic data list components to the page directory since we don't need them anywhere else 

Kudos to @youknowriad  and @ntsekouras for raising most of these improvements

## Why?

1. The image dimensions are constrained and cropped by CSS anyway so, image download size aside, the width and height are not as important since we're not displaying the entire image as we are in the block inspector.
2. The `useSelect` is returning only what the components needs and doesn't return `__()`
3. No premature optimization

## Testing Instructions

There should be no regressions.

1. Activate a block theme
5. Make sure you have some pages ready to role. Add featured images, leave some scheduled too.
6. Now head over to the site editor
7. In the side bar, navigate to Design > Pages and click on the pages.
8. Check each page's details in the sidebar.

## Screenshots
### Trunk

<img width="333" alt="Screenshot 2023-05-30 at 9 25 09 pm" src="https://github.com/WordPress/gutenberg/assets/6458278/98b5f84b-8119-42bc-9621-679731aef067">



### This PR

<img width="337" alt="Screenshot 2023-05-30 at 9 23 57 pm" src="https://github.com/WordPress/gutenberg/assets/6458278/b865af56-03ce-4ad4-8442-126d3c5a6e2c">
